.github/workflows/go.yml

```diff
@@ -394,3 +394,20 @@ jobs:
       - name: Test
         shell: bash
         run: ci/scripts/go_test.sh $(pwd)
+
+  tinygo:
+    name: TinyGo
+    runs-on: ubuntu-latest
+    if: ${{ !contains(github.event.pull_request.title, 'WIP') }}
+    env:
+      TINYGO_VERSION: 0.27.0
```

Comment:
i would suggest adding an image to the docker-compose.yml and a dockerfile to `ci/docker/` which is an image with tinygo installed and then use `archery docker run` like other examples.